lib/rules/validate-order-in-object-keys.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Validates the order in object keys.
+ *
+ * Types: `Boolean` or `String`
+ *
+ * Values:
+ *  - `true` (alias to `asc`)
+ *  - `"asc"`: requires sorting in ascending order
+ *  - `"desc"`: requires sorting in descending order
+ *
+ * #### Example
+ *
+ * ```js
+ * "validateOrderInObjectKeys": "asc"
+ * ```
+ *
+ * ##### Valid
+ *
+ * ```js
+ * var x = {
+ *  x: 'foo',
+ *  y: 'bar'
+ * }
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * var x = {
+ *  y: 'foo',
+ *  x: 'bar'
+ * }
+ * ```
+ */
+
+function asc(a, b) {
+    if (/^[_]/.test(a)) {
```

Comment:
There shouldn't be special cases for underscores. Some people use this as a convention for signifying "private" properties, but some people don't do that. Some people use e.g. "$" and "$$" (angular), others suffix with underscores (node, `util.inherits`). Others don't believe that publicly-accessible properties should even be treated as "private" (because it's not enforced at the language level).

You could instead add a special option for something like this, such as a way for users to specify their own regexp.